src/contracts/libraries/GPv2SafeERC20.sol

```diff
@@ -29,36 +41,58 @@ library GPv2SafeERC20 {
-    /// function does not revert if an address without code is called.
-    ///
-    /// @param token The token targeted by the call.
-    /// @param data The call data (encoded using abi.encode or one of its
-    /// variants).
-    function _callOptionalReturn(IERC20 token, bytes memory data) private {
-        // solhint-disable-next-line avoid-low-level-calls
-        (bool success, bytes memory returndata) = address(token).call(data);
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            let freeMemoryPointer := mload(0x40)
+            mstore(freeMemoryPointer, selector_)
+            mstore(add(freeMemoryPointer, 4), from)
+            mstore(add(freeMemoryPointer, 36), to)
+            mstore(add(freeMemoryPointer, 68), value)
 
-        if (!success) {
-            // Assembly used to revert with correctly encoded error message.
-            // solhint-disable-next-line no-inline-assembly
-            assembly {
-                revert(add(returndata, 0x20), mload(returndata))
+            if iszero(call(gas(), token, 0, freeMemoryPointer, 100, 0, 0)) {
+                returndatacopy(0, 0, returndatasize())
+                revert(0, returndatasize())
             }
         }
-        if (returndata.length > 0) {
-            // Return data is optional
-            require(
-                abi.decode(returndata, (bool)),
-                "GPv2SafeERC20: failed transfer"
-            );
+
+        require(getLastTansferResult(), "GPv2: failed transferFrom");
+    }
+
+    /// @dev Verifies that the last return was a successful `transfer*` call.
+    /// This is done by checking that the return data is either empty, or
+    /// is a valid ABI encoded boolean.
+    function getLastTansferResult() private pure returns (bool) {
+        uint256 success;
+        bool badReturnSize;
+
+        // NOTE: Inspecting previous return data requires assembly. Note that
+        // we write the return data to memory 0 in the case where the return
+        // data size is 32, this is OK since the first 64 bytes of memory are
+        // reserved by Solidy as a scratch space that can be used within
+        // assembly blocks.
+        // <https://docs.soliditylang.org/en/v0.7.6/internals/layout_in_memory.html>
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            switch returndatasize()
+                // Non-standard ERC20 transfer without return.
+                case 0 {
+                    success := 1
+                }
+                // Standard ERC20 transfer returning boolean success value.
+                case 32 {
+                    returndatacopy(0, 0, returndatasize())
+                    success := mload(0)
+                }
+                default {
+                    badReturnSize := 1
+                }
         }
+
+        require(!badReturnSize, "GPv2: malformed transfer result");
+        require(success <= 1, "GPv2: invalid transfer result");
```

Comment:
This line enforces `abicoder v2`, removing this line enforces `abicoder v1`, right?

Reply:
:see_no_evil: Yes, forgot to push my last commit.

Reply:
Last changes look good! Did you choose `iszero(iszero(*))` instead of `gt(*, 0)` because it's cheaper?

Reply:
For a couple of reasons:
1. I've noticed that the generated assembly uses this for `!=0` comparisons
2. Its one byte less bytecode `PUSH1 0 GT` (3 bytes) vs `ISZERO ISZERO` (2 bytes). I think the runtime cost is the same as `PUSH`, `GT` and `ISZERO` are all 3 gas AFAIK.